src/settings-dropdown.jsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from "cockpit";
+import React, { useState } from "react";
+import { CheckIcon, EllipsisVIcon } from "@patternfly/react-icons";
+
+import {
+    Dropdown,
+    DropdownItem,
+    DropdownList,
+    Flex,
+    FlexItem,
+    Icon,
+    MenuToggle
+} from "@patternfly/react-core";
+
+const _ = cockpit.gettext;
+
+export const SettingsDropdown = ({ showHidden, setShowHidden }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const onToggleClick = () => setIsOpen(!isOpen);
+    const onSelect = (_event, _itemId) => setIsOpen(false);
+
+    const onToggleHidden = () => {
+        setShowHidden(prevShowHidden => {
+            localStorage.setItem("cockpit-navigator.showHiddenFiles", !showHidden ? "true" : "false");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-234-20240212-174123-e2d87418-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)